src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -335,10 +335,22 @@ internal void Update(double number)
 
                 case AggregationType.HistogramSumCount:
                     {
-                        lock (this.histogramBuckets.LockObject)
+                        var sw = default(SpinWait);
+                        while (true)
                         {
-                            this.runningValue.AsLong++;
-                            this.histogramBuckets.RunningSum += number;
+                            if (Interlocked.Exchange(ref this.histogramBuckets.IsCriticalSectionOccupied, 1) == 0)
+                            {
+                                unchecked
+                                {
+                                    this.runningValue.AsLong++;
+                                    this.histogramBuckets.RunningSum += number;
+                                }
+
+                                this.histogramBuckets.IsCriticalSectionOccupied = 0;
```

Comment:
I saw this is how the example documentation for [`System.Threading.Interlocked`](https://docs.microsoft.com/en-us/dotnet/api/system.threading.interlocked?view=net-6.0#examples) releases a lock.

Reply:
This is not required. Since the condition to enter the critical section is if the `IsCriticalSectionOccupied == 0` before updating it there will never be two threads that would update the data simultaneously. I'm not sure why that example uses an `Interlocked.Exchange`.